src/WebJobs.Script/Description/DotNet/FunctionSignature.cs

```diff
@@ -18,12 +18,12 @@ namespace Microsoft.Azure.WebJobs.Script.Description
     [CLSCompliant(false)]
     public sealed class FunctionSignature : IEquatable<FunctionSignature>
     {
-        private readonly ImmutableArray<IParameterSymbol> _parameters;
+        private readonly ImmutableArray<FunctionParameter> _parameters;
         private readonly bool _hasLocalTypeReference;
         private readonly string _parentTypeName;
         private readonly string _methodName;
 
-        public FunctionSignature(string parentTypeName, string methodName, ImmutableArray<IParameterSymbol> parameters, bool hasLocalTypeReference)
+        public FunctionSignature(string parentTypeName, string methodName, ImmutableArray<FunctionParameter> parameters, bool hasLocalTypeReference)
```

Comment:
This constructor change and the property on https://github.com/Azure/azure-webjobs-sdk-script/pull/475/files#diff-98d2eb781a41e468e6a39a81a6cab4d4R46 drive most of the changes in this PR and are meant to simplify implementation of other managed languages not supported by Roslyn.